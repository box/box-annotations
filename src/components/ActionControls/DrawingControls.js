// @flow
import React from 'react';

import PlainButton from 'box-react-ui/lib/components/plain-button';
import IconCheck from 'box-react-ui/lib/icons/general/IconCheck';
import IconTrash from 'box-react-ui/lib/icons/general/IconTrash';

import './DrawingControls.scss';

const CLASS_DRAW_CONTROLS = 'ba-action-controls-draw';
const CLASS_DRAWING_SAVE_BTN = 'ba-drawing-save-btn';
const CLASS_DRAWING_DELETE_BTN = 'ba-drawing-delete-btn';

type Props = {
    canDelete: boolean,
    isPending: boolean,
    onCreate: Function,
    onDelete: Function,
};

const DrawingControls = ({ isPending, canDelete, onCreate, onDelete }: Props) => (
    <div className={CLASS_DRAW_CONTROLS}>
        {isPending && (
            <PlainButton className={CLASS_DRAWING_SAVE_BTN} onClick={onCreate} type="button">
                <IconCheck />
            </PlainButton>
        )}
        {canDelete && (
            <PlainButton className={CLASS_DRAWING_DELETE_BTN} onClick={onDelete} type="button">
                <IconTrash height={20} width={20} />
            </PlainButton>
        )}
    </div>
);

DrawingControls.defaultProps = {
    canDelete: false,
};

export default DrawingControls;
