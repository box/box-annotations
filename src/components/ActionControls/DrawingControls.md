### Description
Component to represent the drawing action buttons in the annotation dialog

### Examples
```jsx
const DrawingControlsExamples = require('examples').DrawingControlsExamples;

<DrawingControlsExamples />
```
