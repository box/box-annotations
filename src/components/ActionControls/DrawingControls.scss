@import '../../variables';

.ba {
    .ba-action-controls-draw {
        $button-spacing: 5px;

        display: flex;

        .btn-plain {
            flex-grow: 1;
            padding: 10px;

            .fill-color {
                fill: $sevens;
            }

            &:hover .fill-color {
                fill: $better-black;
            }
        }

        .ba-drawing-save-btn,
        .ba-drawing-save-btn:hover {
            padding-right: $button-spacing;
        }

        .ba-drawing-delete-btn,
        .ba-drawing-delete-btn:hover {
            padding-left: $button-spacing;
        }
    }

    // MOBILE & TABLET CSS
    @media #{$mobile}, #{$tablet} {
        .ba-inline-popover .ba-action-controls-draw {
            background: white;
            border-top: 1px solid $seesee;
            bottom: 0;
            box-shadow: 0 2px 6px 0 rgba(0, 0, 0, .1);
            left: 0;
            margin: 0;
            padding: 15px;
            position: fixed;
            width: 100%;
        }

        .ba-action-controls-draw .btn-plain,
        .ba-action-controls-draw .btn-plain:hover {
            padding: 10px;
        }

        .ba-drawing-save-btn,
        .ba-drawing-save-btn:hover {
            border-right: 1px solid $seesee;
        }
    }
}
