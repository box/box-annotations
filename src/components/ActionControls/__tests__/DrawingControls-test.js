import * as React from 'react';
import { shallow } from 'enzyme';

import DrawingControls from '../DrawingControls';

const onDelete = jest.fn();

describe('components/ActionControls/DrawingControls', () => {
    test('should correctly render pending drawing controls', () => {
        const wrapper = shallow(<DrawingControls canDelete={true} isPending={true} onDelete={onDelete} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should correctly render saved drawing controls', () => {
        const wrapper = shallow(<DrawingControls canDelete={true} onDelete={onDelete} />);
        expect(wrapper).toMatchSnapshot();
    });
});
