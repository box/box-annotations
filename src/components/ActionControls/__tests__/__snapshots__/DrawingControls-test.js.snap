// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ActionControls/DrawingControls should correctly render pending drawing controls 1`] = `
<div
  className="ba-action-controls-draw"
>
  <PlainButton
    className="ba-drawing-save-btn"
    type="button"
  >
    <IconCheck />
  </PlainButton>
  <PlainButton
    className="ba-drawing-delete-btn"
    onClick={[MockFunction]}
    type="button"
  >
    <IconTrash
      height={20}
      width={20}
    />
  </PlainButton>
</div>
`;

exports[`components/ActionControls/DrawingControls should correctly render saved drawing controls 1`] = `
<div
  className="ba-action-controls-draw"
>
  <PlainButton
    className="ba-drawing-delete-btn"
    onClick={[MockFunction]}
    type="button"
  >
    <IconTrash
      height={20}
      width={20}
    />
  </PlainButton>
</div>
`;
